packages/autorest.python/autorest/codegen/__init__.py

```diff
@@ -122,73 +179,65 @@ def remove_cloud_errors(yaml_data: Dict[str, Any]) -> None:
 
     def _build_code_model_options(self) -> Dict[str, Any]:
         """Build en options dict from the user input while running autorest."""
-        azure_arm = self.options.get("azure-arm", False)
-        license_header = self.options.get("header-text", DEFAULT_HEADER_TEXT)
-        if license_header:
-            license_header = license_header.replace("\n", "\n# ")
-            license_header = (
-                "# --------------------------------------------------------------------------\n# "
-                + license_header
-            )
-            license_header += "\n# --------------------------------------------------------------------------"
-
-        low_level_client = cast(bool, self.options.get("low-level-client", False))
-        version_tolerant = cast(bool, self.options.get("version-tolerant", True))
-        show_operations = self.options.get("show-operations", not low_level_client)
-        models_mode_default = (
-            "none" if low_level_client or version_tolerant else "msrest"
-        )
-        if self.options.get("cadl_file") is not None:
-            models_mode_default = "dpg"
 
+        options_retriever = OptionsRetriever(self.options)
         package_name = self.options.get("package-name")
-        unbranded = self.options.get("unbranded", False)
         options: Dict[str, Any] = {
-            "azure_arm": azure_arm,
+            "azure_arm": options_retriever.azure_arm,
             "head_as_boolean": self.options.get("head-as-boolean", True),
-            "license_header": license_header,
+            "license_header": options_retriever.license_header,
```

Comment:
nit: we always access options through the `options_retriever`. Since we don't want to do a property for every single possible flag, you can override the `__attr__` so if there's no property, we just return the flag from the options dict. We should also have a mapping of flag name to default value. We can make this a class variable since it should stay static

```python
class OptionsRetriever:
    OPTIONS_TO_DEFAULT = {
        "head-as-boolean": True,
        "no-async": False,
    }
    def __init__(self, options):
        self.options = options
    ...
    @property
    def company_name(self) -> bool:
        return self.options.get(
            "company-name", "Microsoft" if not self.unbranded else ""
        )

    def __getattr__(self, prop):
...         return self.options.get(prop, self.OPTIONS_TO_DEFAULT[prop])
    

    
```